discretize/utils/meshutils.py

```diff
@@ -255,3 +258,426 @@ def ExtractCoreMesh(xyzlim, mesh, meshType='tensor'):
         raise Exception("Not implemented!")
 
     return actind, meshCore
+
+
+def meshBuilderXYZ(
+    xyz, h,
+    padX=[0, 0], padY=[0, 0], padZ=[0, 0],
+    baseMesh=None,
+    expFact=1.3,
+    meshType='TENSOR',
+    verticalAlignment='top'
+):
+    """
+        Function to quickly generate a Tensor mesh
+        given a cloud of xyz points, finest core cell size
+        and padding distance.
+        If a baseMesh is provided, the core cells will be centered
+        on the underlaying mesh to reduce interpolation errors.
+
+        :param numpy.ndarray xyz: n x 3 array of locations [x, y, z]
+        :param numpy.ndarray h: 1 x 3 cell size for the core mesh
+        :param numpy.ndarray padDist: 2 x 3 padding distances [W,E,S,N,Down,Up]
+        [OPTIONAL]
+        :param numpy.ndarray padCore: Number of core cells around the xyz locs
+        :object SimPEG.Mesh: Base mesh used to shift the new mesh for overlap
+        :param float expFact: Expension factor for padding cells [1.3]
+        :param string meshType: Specify output mesh type: "TensorMesh"
+
+        RETURNS:
+        :object SimPEG.Mesh: Mesh object
+
+    """
+
+    assert meshType in ['TENSOR', 'TREE'], ('Revise meshType. Only ' +
+                                            ' TENSOR | TREE mesh ' +
+                                            'are implemented')
+
+    assert verticalAlignment in ['center', 'top'], ("verticalAlignment must be 'center' | 'top'")
+
```

Comment:
Replace `assert` with an `if`-statement and `raise`. `assert` should only be used for testing, as certain flags will remove or ignore `assert` statements.